include/cglm/quat.h

```diff
@@ -184,10 +187,40 @@ glm_quat_copy(versor q, versor dest) {
   glm_vec4_copy(q, dest);
 }
 
+/*!
+ * @brief compute quaternion rotating vector A to vector B
+ *
+ * @param[in]   a     vec3 (must have unit length)
+ * @param[in]   b     vec3 (must have unit length)
+ * @param[out]  dest  quaternion (of unit length)
+ */
+CGLM_INLINE
+void
+glm_quat_from_vecs(vec3 a, vec3 b, versor dest) {
+  float cos_theta = glm_vec3_dot(a, b);
+  if (cos_theta >= 1.f - GLM_FLT_EPSILON) {  // a ∥ b
+    glm_quat_identity(dest);
+    return;
+  }
+  CGLM_ALIGN(8) vec3 axis;
+  float cos_half_theta;
+  if (cos_theta < -1.f + GLM_FLT_EPSILON) {  // angle(a, b) = 180°
+    glm_vec3_ortho(a, axis);
+    cos_half_theta = 0.f;
+  }
+  else {
+    glm_vec3_cross(a, b, axis);
+    const float cos_zero = 1.0f;
```

Comment:
I think there is no need to create separate consst variable. Float literal woul also be a constant: `1.0f + cos_theta`